types/module/simulation.go

```diff
@@ -22,7 +21,7 @@ type AppModuleSimulation interface {
 	ProposalContents(simState SimulationState) []simulation.WeightedProposalContent
 
 	// register a func to decode the each module's defined types from their corresponding store key
-	RegisterStoreDecoder(storetypes.StoreDecoderRegistry)
```

Comment:
this is interface breaking but I think its acceptable since other wise store has a simulation type for no reason